zypp/solver/detail/SATResolver.cc

```diff
@@ -178,6 +179,43 @@ int vendorCheck( sat::detail::CPool *pool, Solvable *solvable1, Solvable *solvab
                                             IdString(solvable2->vendor) ) ? 0 : 1;
 }
 
+/** ResPool helper to compute the initial status of Patches etc.
+ * An empty solver run (no jobs) just to compute the initial status
+ * of pseudo installed items (patches).
+ */
+void establish( sat::Queue & pseudoItems_r, sat::Queue & pseudoFlags_r )
+{
+  pseudoItems_r = collectPseudoInstalled( ResPool::instance() );
+  if ( ! pseudoItems_r.empty() )
+  {
+    MIL << "Establish..." << endl;
+    sat::detail::CPool * cPool { sat::Pool::instance().get() };
+    ::pool_set_custom_vendorcheck( cPool, &vendorCheck );
+
+    sat::Queue jobQueue;
+    AutoDispose<sat::detail::CSolver*> cSolver { ::solver_create( cPool ), ::solver_free };
+    sat::Pool::instance().prepare();
+    if ( ::solver_solve( cSolver, jobQueue ) != 0 )
+      INT << "How can establish fail?" << endl;
```

Comment:
After a error like that does it make sense to even continue here?

Reply:
There are no jobs, so the error shouldn't happen (would be kind of 'internal solver error'). Just to be sure.